### DISCUSSION TITLE
How to install this package in Gentoo?

### DISCUSSION THREAD

**Discussion by mak163** [abraunegg/onedrive#1937] [Q&A] [answered]
Unable to install this package in Gentoo.
Completed all points from Wiki https://wiki.dlang.org/Installing_DMD
```
emerge app-portage/layman
layman -f -a dlang
emerge dev-lang/dmd
```
Installing a package and getting an error
```
# emerge --ask net-misc/onedrive

These are the packages that would be merged, in order:

Calculating dependencies /

!!! Problem resolving dependencies for net-misc/onedrive
... done!

!!! The ebuild selected to satisfy "net-misc/onedrive" has unmet requirements.
- net-misc/onedrive-2.4.16::dlang USE="debug dmd-2_087 dmd-2_088 dmd-2_089 dmd-2_090 dmd-2_091 dmd-2_092 dmd-2_093 dmd-2_094 dmd-2_095 dmd-2_096 ldc2-1_25 ldc2-1_26 ldc2-1_27 ldc2-1_28 libnotify" ABI_X86="(64)"

  The following REQUIRED_USE flag constraints are unsatisfied:
    exactly-one-of ( dmd-2_089 dmd-2_088 dmd-2_087 dmd-2_090 dmd-2_091 dmd-2_092 dmd-2_093 dmd-2_094 dmd-2_095 dmd-2_096 ldc2-1_28 ldc2-1_26 ldc2-1_27 ldc2-1_25 )
```
What should I do to install the package?

**Comment by abraunegg** [comment, marked as answer]
@mak163 
Unfortunately I am unable to assist here as this is something unique to Gentoo .. suggest you ask on the [Gentoo Forums](https://forums.gentoo.org/) for assistance and then potentially circle back here with your answer ...

I know this is not the answer you are looking for - however I have zero knowledge for Gentoo.

**Reply by mak163** [reply]
The problem was solved very simply. It was necessary to add the USE flag to the package description
`nano /etc/portage/package.use/onedrive
net-misc/onedrive dmd-2_096
`

**Reply by mak163** [reply]
But now another problem has arisen. Can't run as a service
The attached script `/etc/init.d/onedrive` is not suitable for Open RC

**Comment by abraunegg** [comment]
@mak163 
> But now another problem has arisen. Can't run as a service The attached script `/etc/init.d/onedrive` is not suitable for Open RC


This is because this init script was written for CentOS 6 - not OpenRC. The existing init script needs to be depreciated and removed from the repository.

Unfortunately - again, there is no ideal solution for you using Gentoo. Please feel free to create an applicable OpenRC script and then contribute that as a PR for inclusion.
